contract/p/gnoswap/version_manager/version_manager_test.gno

```diff
@@ -1051,184 +753,29 @@ func TestChangeImplementation_StateConsistency(t *testing.T) {
 			}
 
 			// Verify implementation is not nil
-			if vmInternal.GetCurrentImplementation() == nil {
-				t.Error("implementation should not be nil after changes")
-			}
-		})
-	}
-}
-
-// TestRegisterInitializer_ErrorBranches tests all error branches in RegisterInitializer
-func TestRegisterInitializer_ErrorBranches(t *testing.T) {
-	tests := []struct {
-		name          string
-		description   string
-		setupFunc     func() (*versionManager, error)
-		errorContains string
-	}{
-		{
-			name:        "duplicate registration error",
-			description: "RegisterInitializer should return error when initializer already registered",
-			setupFunc: func() (*versionManager, error) {
-				kvStore := store.NewKVStore(mockDomainAddr)
-				vm := &versionManager{
-					domainPath:              "gno.land/r/gnoswap/protocol_fee",
-					domainKVStore:           kvStore,
-					initializeDomainStoreFn: mockInitializeDomainStoreFn,
-					initializers:            avl.NewTree(),
-					currentPackagePath:      "",
-					currentImplementation:   nil,
-				}
-
-				// Pre-register an initializer
-				vm.initializers.Set("gno.land/r/gnoswap/protocol_fee/v1", mockInitializer("v1"))
-
-				// Simulate checking for duplicate
-				if vm.initializers.Has("gno.land/r/gnoswap/protocol_fee/v1") {
-					return vm, errors.New("version_manager: initializer already registered")
-				}
-				return vm, nil
-			},
-			errorContains: "initializer already registered",
-		},
```

Comment:
These tests were only simulating conditions without actually calling `RegisterInitializer`. Since `RegisterInitializer` uses `runtime.PreviousRealm()`, there is an issue where the "caller cannot be user" error occurs first in unit tests, making it impossible to reach the check logic.

Because simulation tests can pass even when production code changes, they produce false-positive results. Therefore, the problematic cases were removed to prevent regression issues.